### PR TITLE
Add remove command

### PR DESCRIPTION
**Pull request by L1uY1jun**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -96,6 +96,21 @@ public static ApplicationStatus parseApplicationStatus(String applicationStatus)
...
+     * Parses a {@code String applicationStatus} into a completed {@code ApplicationStatus}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code applicationStatus} is invalid or pending.
+     */
+    public static ApplicationStatus parseCompletedApplicationStatus(String applicationStatus) throws ParseException {
+        requireNonNull(applicationStatus);
+        String trimmedApplicationStatus = applicationStatus.trim();
+        if (!ApplicationStatus.isCompletedApplicationStatus(trimmedApplicationStatus)) {
+            throw new ParseException(ApplicationStatus.MESSAGE_COMPLETED_STATUS_CONSTRAINTS);
```
Comment (benjytan45678) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Not sure whether MESSAGE_COMPLETED_STATUS_CONSTRAINTS is a good name. Maybe can switch to MESSAGE_STATUS_REJECTION

File: src/main/java/seedu/address/model/applicant/UniqueApplicantList.java [hunk 2]
```diff
@@ -79,6 +79,18 @@ public void remove(Applicant toRemove) {
         }
     }
 
+    /**
+     * Removes all applicants with matching application status from the list.
+     */
+    public void removeByStatus(ApplicationStatus statusToRemove) {
+        requireNonNull(statusToRemove);
+        for (int index = internalList.size() - 1; index >= 0; index--) {
+            if (internalList.get(index).isMatchingApplicationStatus(statusToRemove)) {
```
Comment (benjytan45678) [inline src/main/java/seedu/address/model/applicant/UniqueApplicantList.java:89]: Maybe violates law of demeter here. Can try to create a new method for this?

Comment (L1uY1jun) [reply src/main/java/seedu/address/model/applicant/UniqueApplicantList.java:89]: the internalList is an attribute of UniqueApplicantList though so i don't think it violates the law of demeter here. There is also no other way to check into the internalList as ObservableList<E> does not provide any functionality.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (L1uY1jun) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Alright I have changed it to better fit the context

File: src/main/java/seedu/address/model/applicant/UniqueApplicantList.java (hunk 2 above)
Comment (benjytan45678) [reply src/main/java/seedu/address/model/applicant/UniqueApplicantList.java:89]: okay

**Pull request merged**